django/core/paginator.py

```diff
@@ -96,19 +95,35 @@ def _check_object_list_is_ordered(self):
         """
         Warn if self.object_list is unordered (typically a QuerySet).
         """
-        if hasattr(self.object_list, 'ordered') and not self.object_list.ordered:
+        if hasattr(self.object_list,
+                   'ordered') and not self.object_list.ordered:
             warnings.warn(
                 'Pagination may yield inconsistent results with an unordered '
                 'object_list: {!r}'.format(self.object_list),
                 UnorderedObjectListWarning
             )
 
+    def page_force_into_range(self, number):
+        """
+        Always return a valid page, even if the page argument is not in range
+        or is invalid.  If no objects and allow_empty_first_page=False, this
+        will still throw an exception.
+        """
+        # This utility method implements the boilerplate code
+        # in previous Django versions documentation.
+        try:
+            number = self.validate_number(number)
+        except PageNotAnInteger:
+            number = 1
+        except EmptyPage:
+            number = self.num_pages
+        return self.page(number)
```

Comment:
It looks like this is showing up as a requested change, but I don't think that was @MarkusH's intent?